tests/unit/index.ts

```diff
@@ -82,12 +82,37 @@ registerSuite({
 			assert.include(contents, `from 'sub/baz';`);
 		});
 	},
+	'test prefixing of absolute paths with options.name': function () {
+		// we reuse foo-directories from the directory stress test above.
+		// not sure if this is the best idea because it means these tests aren't
+		// totally orthogonal, but it doesn't make the test slower.
+		return generate({
+			project: 'tests/support/foo-directories',
+			out: 'tmp/foo.config.d.ts',
+			main: '__abs_prefix/sub/index',
+			name: '__abs_prefix'
+		}).then(function () {
+			const contents = fs.readFileSync('tmp/foo.config.d.ts', { encoding: 'utf8' });
+			assert(contents, 'foo.config.d.ts should exist and have contents');
+			assert.include(contents, `module '__abs_prefix/sub/index'`);
+			assert.include(contents, `module '__abs_prefix/sub/Bar'`);
+			assert.include(contents, `module '__abs_prefix/sub/baz'`);
+
+			// also check imports look right
+			assert.include(contents, `import Bar from '__abs_prefix/sub/Bar'`);
+			assert.include(contents, `from '__abs_prefix/sub/baz';`);
+
+			// and look at the generated main code
+			assert.include(contents, `module '__abs_prefix'`);
+			assert.include(contents, `import main = require('__abs_prefix/sub/index')`);
+		});
+	},
 	'es6 main module': function () {
 		return generate({
 			name: 'foo',
 			project: 'tests/support/foo-es6',
 			out: 'tmp/foo.es6.d.ts',
-			main: 'index.ts'
+			main: 'foo/index'
```

Comment:
this is necessary because of the added input validation - main is supposed to reference the module identifier of one of the modules we output, and index.ts is certainly not a module id but a file.